packages/core/src/util/packageInfo.ts

```diff
@@ -0,0 +1,22 @@
+import findup from 'find-up';
+import path from 'node:path';
+
+export function findClosestPackageJson(file: string): string | undefined {
+  return findup.sync('package.json', { cwd: path.dirname(file) });
+}
+
+export function getPackageJsonPath(): string {
+  const packageJsonPath = findClosestPackageJson(__filename);
+
+  if (!packageJsonPath) {
+    throw new Error('not found package.json');
```

Comment:
The message could be a bit detailed and `not found package.json` should be `package.json not found`